From 3928a33094d9d5c224c3d0a4d428edb85e26096d Mon Sep 17 00:00:00 2001
From: Sanyog Kale <skale@habana.ai>
Date: Fri, 19 Aug 2022 20:23:10 +0300
Subject: [PATCH] [SW-86869] common: modify logic for computing min fw version

Current logic allows to load minimum firmware version 0 and 1 if
0th bit of firmware version is burned in efuse. firmware version 0 is
pre-production version and should not be allowed to load if 0th bit
is burned (ie minimum firmware version 1) in efuse. This patch updates
logic to allow only minimum firmware version 1 if 0th bit is burned.
This also means the allowed production minimum version range is
from 1 to 128.

Change-Id: Iff83e732bdf746b2955913b454866fbbb655525a
---

diff --git a/common/security.c b/common/security.c
index e5fe0b9..0eccc43 100644
--- a/common/security.c
+++ b/common/security.c
@@ -81,9 +81,9 @@
 		for (colume = SECURITY_FW_VER_LAST_BYTE; colume >= 0;
 				colume--) {
 			/* find highest bit set in valid FW */
-			fw_version_sec--;
 			if (fw_version[row] & BIT(colume))
 				return fw_version_sec;
+			fw_version_sec--;
 		}
 	}
 
